scipy/_build_utils/_generate_blas_wrapper.py

```diff
@@ -0,0 +1,126 @@
+"""
+Generate wrappers to dispatch BLAS/LAPACK calls to the properly prefixed/
+suffixed symbols.
+
+For example, MacOS 13.3+ has a new, LAPACK 3.9-compatible, high performance
+BLAS/LAPACK implementation. These functions are provided side-by-side with
+the old implementation, and the symbols are distinguished by appending the
+literal suffix "$NEWLAPACK".
+
+To point our BLAS/LAPACK calls to these symbols, we need to create wrappers
+which call them appropriately. We do this as simple C function declarations
+that make use of the preprocessor macros defined in npy_cblas.h.
+
+We already have all the required signature information in
+    scipy/linalg/cython_{blas,lapack}_signatures.txt
+which is generated by
+    scipy/linalg/_cython_signature_generator.py
+
+We automatically create the declarations based on these signatures, with a
+few special cases. First, all complex-valued functions are skipped (empty
+source files) because they require more complicated wrapper logic. The
+wrappers for these functions are hard-coded in wrap_g77_abi.c and
+wrap_dummy_g77_abi.c. Second, certain functions are missing from the
+new Accelerate implementation and/or have unusual symbols that require
+special handling in this script.
+"""
+import argparse
+import os
+
+from _wrappers_common import (C_PREAMBLE, C_TYPES, CPP_GUARD_BEGIN,
+                              CPP_GUARD_END, LAPACK_DECLS, WRAPPED_FUNCS,
+                              all_newer, get_blas_macro_and_name,
+                              read_signatures, write_files)
+
+CURR_DIR = os.path.dirname(os.path.abspath(__file__))
+LINALG_DIR = os.path.abspath(os.path.join(CURR_DIR, "..", "linalg"))
+comments = f"""/*
+This file was generated by {os.path.basename(__file__)}.
+Do not edit this file directly.
+*/"""
+
+
+def generate_decl_wrapper(name, return_type, argnames, argtypes, accelerate):
+    """
+    Create the main body of the wrapper file.
+
+    Wrapper has symbol `F_FUNC(name,NAME)` and wraps the BLAS/LAPACK function
+    `blas_macro(blas_name)` (by default: `BLAS_FUNC(name)`).
+    """
+    # Complex-valued functions have hard-coded wrappers in G77 ABI wrappers
+    if name in WRAPPED_FUNCS:
+        return ""
+    # lsame and dcabs1 are missing from new Accelerate
+    # Use old symbols that do not require a wrapper
+    if accelerate and name in ['lsame', 'dcabs1']:
+        return ""
+    c_return_type = C_TYPES[return_type]
+    c_argtypes = [C_TYPES[t] for t in argtypes]
+    param_list = ', '.join(f'{t} *{n}' for t, n in zip(c_argtypes, argnames))
+    argnames = ', '.join(argnames)
+    blas_macro, blas_name = get_blas_macro_and_name(name, accelerate)
+    return f"""
+{c_return_type} {blas_macro}({blas_name})({param_list});
+{c_return_type} F_FUNC({name},{name.upper()})({param_list}){{
+    return {blas_macro}({blas_name})({argnames});
+}}
+"""
+
+
+def generate_file_wrapper(sigs, lib_name, accelerate, outdir):
```

Comment:
Let's add a short docstring, and that this returns a dictionary of `{file_path: file_content}`.